src/App.jsx

```diff
@@ -14,33 +16,93 @@ export const goodsFromServer = [
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD = {
+  alphabeticaly: 'alphabetic',
+  length: 'length',
+};
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  function sortGoods(input, sortField, reverse) {
+    const preparedGoods = [...input];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    if (sortField) {
+      preparedGoods.sort((good1, good2) => {
+        switch (sortField) {
+          case SORT_FIELD.alphabeticaly:
+            return good1.localeCompare(good2);
+          case SORT_FIELD.length:
+            return good1.length - good2.length;
+          default:
+            return 0;
+        }
+      });
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (reverse === 'desc') {
+      preparedGoods.reverse();
+    }
+
+    return preparedGoods;
+  }
+
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState('asc');
+  const visibleGoods = sortGoods(goodsFromServer, sortField, reverse);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SORT_FIELD.alphabeticaly,
+          })}
+          onClick={() => setSortField(SORT_FIELD.alphabeticaly)}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SORT_FIELD.length,
+          })}
+          onClick={() => setSortField(SORT_FIELD.length)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': reverse === 'asc',
+          })}
+          onClick={() => {
+            setReverse(reverse === 'asc' ? 'desc' : 'asc');
+          }}
+        >
+          Reverse
+        </button>
+
+        {(sortField || reverse === 'desc') && (
+          <button
+            onClick={() => {
+              setSortField('');
+              setReverse('asc');
+            }}
+            type="button"
+            className="button is-danger is-light"
+          >
+            Reset
+          </button>
+        )}
+      </div>
+
+      <ul className="goodList">
+        {visibleGoods.map(good => {
+          return <li data-cy="Good">{good}</li>;
```

Comment:
The `data-cy` attribute is used for testing purposes and should have unique values for each list item if you intend to use it for testing individual items. Consider adding a unique identifier to each `data-cy` attribute.